packages/mobile/src/account/DataSaver.tsx

```diff
@@ -122,48 +122,52 @@ export class DataSaver extends React.Component<Props, State> {
   }
 
   onPressToggleWithSwitchOnModal = () => {
-    this.props.toggleZeroSyncMode(true)
+    this.props.toggleFornoMode(true)
     this.hideSwitchOnModal()
   }
 
   onPressPromptModal = () => {
-    this.props.toggleZeroSyncMode(true)
+    this.props.toggleFornoMode(true)
     navigateBack()
   }
 
   hidePromptModal = () => {
-    this.props.toggleZeroSyncMode(false)
+    this.props.toggleFornoMode(false)
     navigateBack()
   }
 
-  handleZeroSyncToggle = (zeroSyncMode: boolean) => {
-    if (!zeroSyncMode && this.props.gethStartedThisSession) {
-      // Starting geth a second time this app session which will
-      // require an app restart, so show restart modal
-      this.showSwitchOffModal()
+  handleFornoToggle = (fornoMode: boolean) => {
```

Comment:
Fixes if else- if switching off forno mode, show the modal if need be, else just turn it off. If switching on forno mode, show switch on modal. 

Previously, if switching off forno mode and modal not necessary, this would switch forno on (opposite of intended behavior)